mypyc/emitmodule.py

```diff
@@ -244,34 +312,66 @@ def generate_shared_lib_init(self, emitter: Emitter) -> None:
              .format(self.shared_lib_name)),
             'int res;',
             'PyObject *capsule;',
-            'PyObject *module = PyModule_Create(&def);',
+            'static PyObject *module;',
+            'if (module) {',
+            'Py_INCREF(module);',
+            'return module;',
+            '}',
+            'module = PyModule_Create(&def);',
             'if (!module) {',
             'goto fail;',
             '}',
             '',
         )
 
+        emitter.emit_lines(
+            'capsule = PyCapsule_New(&exports, "{}.exports", NULL);'.format(
+                self.shared_lib_name),
+            'if (!capsule) {',
+            'goto fail;',
+            '}',
+            'res = PyObject_SetAttrString(module, "exports", capsule);',
```

Comment:
Does this get added to the module namespace? What if there's an attribute with name `exports`?

Reply:
This is part of the module namespace for the shared library module, not the actual module.